pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -0,0 +1,520 @@
+	destinationTransportPort             uint16
+	protocolIdentifier                   uint8
+	packetTotalCount                     uint64
+	octetTotalCount                      uint64
+	packetDeltaCount                     uint64
+	octetDeltaCount                      uint64
+	reversePacketTotalCount              uint64
+	reverseOctetTotalCount               uint64
+	reversePacketDeltaCount              uint64
+	reverseOctetDeltaCount               uint64
+	sourcePodName                        string
+	sourcePodNamespace                   string
+	sourceNodeName                       string
+	destinationPodName                   string
+	destinationPodNamespace              string
+	destinationNodeName                  string
+	destinationClusterIP                 string
+	destinationServicePort               uint16
+	destinationServicePortName           string
+	ingressNetworkPolicyName             string
+	ingressNetworkPolicyNamespace        string
+	ingressNetworkPolicyRuleName         string
+	ingressNetworkPolicyRuleAction       uint8
+	ingressNetworkPolicyType             uint8
+	egressNetworkPolicyName              string
+	egressNetworkPolicyNamespace         string
+	egressNetworkPolicyRuleName          string
+	egressNetworkPolicyRuleAction        uint8
+	egressNetworkPolicyType              uint8
+	tcpState                             string
+	flowType                             uint8
+	sourcePodLabels                      string
+	destinationPodLabels                 string
+	throughput                           uint64
+	reverseThroughput                    uint64
+	throughputFromSourceNode             uint64
+	throughputFromDestinationNode        uint64
+	reverseThroughputFromSourceNode      uint64
+	reverseThroughputFromDestinationNode uint64
+}
+
+func NewClickHouseClient(input ClickHouseInput) (*ClickHouseExportProcess, error) {
+	dsn, err := input.getDataSourceName()
+	if err != nil {
+		klog.ErrorS(err, "Error parsing ClickHouse DSN")
+		return nil, err
+	}
+	connect, err := sql.Open("clickhouse", dsn)
+	if err != nil {
+		klog.ErrorS(err, "Error when opening DB connection", "dbURL", input.DatabaseURL)
+		return nil, err
+	}
+
+	// Test ping DB
+	if err = connect.Ping(); err != nil {
+		if exception, ok := err.(*clickhouse.Exception); ok {
+			klog.ErrorS(err, "Error pinging DB", "errCode", exception.Code)
+		} else {
+			klog.ErrorS(err, "Error pinging DB")
+		}
+		return nil, err
+	}
+
+	// Test open Transaction
+	tx, err := connect.Begin()
+	if err == nil {
+		_, err = tx.Prepare(insertQuery)
+	}
+	if err != nil {
+		klog.ErrorS(err, "Error when preparing insert statement")
+		return nil, err
+	}
+	_ = tx.Commit()
+
+	chClient := &ClickHouseExportProcess{
+		db:             connect,
+		dsn:            dsn,
+		deque:          deque.New(),
+		mutex:          sync.RWMutex{},
+		stopChan:       make(chan bool),
+		commitInterval: input.CommitInterval,
+	}
+	chClient.queueSize = maxQueueSize
```

Comment:
why not do this initialization inside of the `ClickHouseExportProcess{...}`?

Reply:
fixed. was previously left out for a hack in UT.